corfudb-tools/src/main/java/org/corfudb/compactor/CorfuStoreCompactorMain.java

```diff
@@ -0,0 +1,412 @@
+            "[--tlsEnabled=<tls_enabled>]\n"
+            + "Options:\n"
+            + "--hostname=<hostname>   Hostname\n"
+            + "--port=<port>   Port\n"
+            + "--keystore=<keystore_file> KeyStore File\n"
+            + "--ks_password=<keystore_password> KeyStore Password\n"
+            + "--truststore=<truststore_file> TrustStore File\n"
+            + "--truststore_password=<truststore_password> Truststore Password\n"
+            + "--persistedCacheRoot=<pathToTempDirForLargeTables> Path to Temp Dir\n"
+            + "--maxWriteSize=<maxWriteSize> Max write size smaller than 2GB\n"
+            + "--bulkReadSize=<bulkReadSize> Read size for chain replication\n"
+            + "--trim=<trim> Should trim be performed in this run\n"
+            + "--isUpgrade=<isUpgrade> Is this called during upgrade\n"
+            + "--upgradeDescriptorTable=<upgradeDescriptorTable> Repopulate descriptor table?\n"
+            + "--keyToDelete=<keyToDelete> Key of the record to be deleted\n"
+            + "--tlsEnabled=<tls_enabled>";
+
+    public static void main(String[] args) throws Exception {
+        CorfuStoreCompactorMain corfuCompactorMain = new CorfuStoreCompactorMain();
+        corfuCompactorMain.getCompactorArgs(args);
+
+        CorfuRuntimeHelper corfuRuntimeHelper;
+        if (maxWriteSize == -1) {
+            if (persistedCacheRoot == null) {
+                // in-memory compaction
+                maxWriteSize = DEFAULT_CP_MAX_WRITE_SIZE;
+            } else {
+                // disk-backed non-config compaction
+                maxWriteSize = NON_CONFIG_DEFAULT_CP_MAX_WRITE_SIZE;
+            }
+        }
+        if (tlsEnabled) {
+            corfuRuntimeHelper = new CorfuRuntimeHelper(hostname, port, maxWriteSize, bulkReadSize,
+                    runtimeKeyStore, runtimeKeystorePasswordFile,
+                    runtimeTrustStore, runtimeTrustStorePasswordFile);
+        } else {
+            corfuRuntimeHelper = new CorfuRuntimeHelper(hostname, port, maxWriteSize, bulkReadSize);
+        }
+
+        corfuRuntime = corfuRuntimeHelper.getRuntime();
+        corfuCompactor = new CorfuStoreCompactor(corfuRuntime, trim, persistedCacheRoot);
+        thisNodeUuid = UUID.fromString(CorfuRuntimeHelper.getThisNodeUuid());
+
+        if (isCheckpointFrozen(corfuRuntime)) {
+            return;
+        }
+
+        if (trim) {
+            trimAndUpdateToken();
+        }
+
+        if (isUpgrade) {
+            log.info("Upgrade: Saving Trim Token");
+
+            if (upgradeDescriptorTable) {
+                log.info("Upgrade to CorfuStore v2: Deleting components inside checkpoint map except UFO");
+                final CorfuTable<String, Token> checkpointMap = CorfuRuntimeHelper.getCheckpointMap(corfuRuntime);
+                Set<String> keySet = checkpointMap.keySet();
+                for (String key : keySet) {
+                    if (!key.equals("ufo")) {
```

Comment:
Need to change the key to something else.  Maybe CheckpointToken?

Reply:
this whole checkpointMap needs to be gone i think